esmvalcore/preprocessor/_regrid.py

```diff
@@ -650,11 +651,38 @@ def regrid(cube, target_grid, scheme, lat_offset=True, lon_offset=True):
         # scheme is a function f(src_cube, grid_cube) -> Cube
         cube = loaded_scheme
     else:
+        cube = _rechunk(cube, target_grid)
         cube = cube.regrid(target_grid, loaded_scheme)
 
     return cube
 
 
+def _rechunk(cube, target_grid):
+    """Re-chunk cube with optimal chunk sizes for target grid."""
+    if not cube.has_lazy_data() or cube.ndim < 3:
+        # Only rechunk lazy multidimensional data
+        return cube
+
+    if 2 * np.prod(cube.shape[-2:]) > np.prod(target_grid.shape):
+        # Only rechunk if target grid is more than a factor of 2 larger,
+        # because rechunking will keep the original chunk in memory.
+        return cube
+
+    data = cube.lazy_data()
+
+    # Compute a good chunk size for the target array
+    tgt_shape = data.shape[:-2] + target_grid.shape
+    tgt_chunks = data.chunks[:-2] + target_grid.shape
+    tgt_data = da.empty(tgt_shape, dtype=data.dtype, chunks=tgt_chunks)
+    tgt_data = tgt_data.rechunk({i: "auto" for i in range(cube.ndim - 2)})
```

Comment:
dask rechunking is a bit slow and bulky, but I don't know actual performance parameters for it, unfortunately - are we sure this adds a benefit and doesn't just make things slower?

Reply:
Yes, it reduces memory consumption: if you're computing multimodel statistics you will load the n_datasets x chunksize into memory of a worker, which may be much already. If you combine that with regridding to a higher resolution grid, it really blows up memory use. Maybe we should do something similar for the multimodel statistics too. If I remember correctly, we even had code to do that in this pull request at some point.

Reply:
Note that it's only rechunking if the regridded chunks are at least twice as big as the input chunks (see also explanation on line 667).

Reply:
yes, that is a very good point that I completely missed, cheers, bud! :beer: